### ISSUE TITLE
libignition-common3 / libignition-cmake2 cannot find AVDEVICE without pkg-config installed

### ISSUE THREAD

**Issue by Tobias-Fischer** [conda-forge/libignition-common-feedstock#58]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Hi @traversaro,

In the robostack-humble rebuild, I ran into the following error on linux-64 (see https://github.com/RoboStack/ros-humble/actions/runs/3652312631/jobs/6172952920):
```
CMake Error at /opt/conda/build_artifacts/ros-humble-ros-ign-gazebo_1670547596520/_build_env/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find AVDEVICE (missing: AVDEVICE_FOUND) (Required is at least
  version "56.4.100")
Call Stack (most recent call first):
  /opt/conda/build_artifacts/ros-humble-ros-ign-gazebo_1670547596520/_build_env/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /opt/conda/build_artifacts/ros-humble-ros-ign-gazebo_1670547596520/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/share/cmake/ignition-cmake2/cmake2/FindAVDEVICE.cmake:53 (find_package_handle_standard_args)
  /opt/conda/build_artifacts/ros-humble-ros-ign-gazebo_1670547596520/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/cmake/ignition-common4-av/ignition-common4-av-config.cmake:105 (find_package)
  /opt/conda/build_artifacts/ros-humble-ros-ign-gazebo_1670547596520/_build_env/share/cmake-3.23/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/conda/build_artifacts/ros-humble-ros-ign-gazebo_1670547596520/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/cmake/ignition-common4/ignition-common4-config.cmake:187 (find_dependency)
  /opt/conda/build_artifacts/ros-humble-ros-ign-gazebo_1670547596520/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/cmake/ignition-gazebo6/ignition-gazebo6-config.cmake:96 (find_package)
  CMakeLists.txt:41 (find_package)
```

I think this is because of a change in ffmpeg, where they split version.h and version_major.h: https://github.com/FFmpeg/FFmpeg/blob/master/libavdevice/version.h

In libignition-cmake, it tries to find the version purely based on version.h: https://github.com/gazebosim/gz-cmake/blob/ign-cmake2/cmake/FindAVDEVICE.cmake

My best bet is that this needs updating. As you have recently looked at the FindAVDEVICE @traversaro, it would be great if you could take a look.

Many thanks!

### Installed packages

```shell
+ libignition-cmake2                                       2.16.0  h27087fc_0           conda-forge/linux-64           Cached
  + libignition-common4                                       4.6.1  h24a332d_0           conda-forge/linux-64           Cached
  + ffmpeg                                                    5.1.2  gpl_hc51e5dc_103     conda-forge/linux-64           Cached
```

### Environment info

```shell
n/a
```

**Comment by traversaro** [comment]
I think this is indeed a bug, but a bug of a specific configuration, i.e. finding ffmpeg **without pkg-config** installed. I added that configuration in https://github.com/gazebosim/gz-cmake/pull/95, as back in time vcpkg did not used/installed pkg-config. However, now vcpkg installs pkg-config, so the configuration is not really tested a lot. I will look into this, but just to be sure it would be a good idea to always install pkg-config when cmake is installed. I did not add it as a run export to this package, even if we could consider adding it as a run dependency to ign-cmake or ign-common, as it is not how tipically cmake or pkg-config dependency are handled, but we could consider that.
